Automattic/newspack-theme
# chore: update build-related packages

### All Submissions:






### Changes proposed in this Pull Request:



Updates Composer and NPM packages related to asset building and linting, and fixes auto-formatting using Prettier. Most other Newspack repos already support auto-formatting of code using Prettier, but this one didn't have the correct WP coding standards implemented in the repo, so you needed to pull in the standards files externally. After this, it should be possible to run `npm run format:js` and `npm run format:scss` or enable auto-formatting in your IDE via Prettier without needing to do any additional config.

Note that the package versions we're updating to here aren't necessarily the latest versions available, just the latest versions we've already implemented in other Newspack repos (to hopefully minimize the amount of testing needed).

### How to test the changes in this Pull Request:

There shouldn't be any functional changes, but you can test by cloning this repo to a fresh directory, then:

1. Check out this branch and run `composer install && npm ci`
2. Run `npm run format:js` and `npm run format:scss`, and confirm that all source JS and SCSS files are auto-formatted according to WP coding standards.
3. If your IDE is set to auto-format via Prettier, edit a file in `/js/src` with some non-standards-compliant JS, like `console.log('not standard')` and trigger the auto-format. The JS should be auto-formatted to WP standards, e.g. `console.log( 'not standard' );` (note spaces in parentheses and trailing semicolon).

### Other information:
